api/server/server.go

```diff
@@ -81,55 +93,83 @@ type Config struct {
 
 // NewServer starts and returns a new server with the given configuration.
 func NewServer(conf Config) (*Server, error) {
-	c, cls, err := lotus.New(conf.LotusAddress, conf.LotusAuthToken)
+	var c *apistruct.FullNodeStruct
+	var cls func()
+	var err error
+	var masterAddr address.Address
+	if conf.Embedded {
+		c, cls, err = lotus.NewEmbedded()
+		masterAddr, err = c.WalletDefaultAddress(context.Background())
+		if err != nil {
+			return nil, fmt.Errorf("getting default address: %s", err)
+		}
+
+	} else {
+		c, cls, err = lotus.New(conf.LotusAddress, conf.LotusAuthToken)
+	}
 	if err != nil {
 		return nil, err
 	}
 
 	fchost, err := fchost.New()
 	if err != nil {
-		return nil, fmt.Errorf("error when creating filecoin host: %s", err)
+		return nil, fmt.Errorf("creating filecoin host: %s", err)
 	}
 	if err := fchost.Bootstrap(); err != nil {
-		return nil, fmt.Errorf("error when bootstrapping filecoin host: %s", err)
+		return nil, fmt.Errorf("bootstrapping filecoin host: %s", err)
 	}
 
 	path := filepath.Join(conf.RepoPath, datastoreFolderName)
 	if err := os.MkdirAll(path, os.ModePerm); err != nil {
-		return nil, fmt.Errorf("error when creating repo folder: %s", err)
+		return nil, fmt.Errorf("creating repo folder: %s", err)
 	}
 
 	ds, err := badger.NewDatastore(path, &badger.DefaultOptions)
 	if err != nil {
-		return nil, fmt.Errorf("error when opening datastore on repo: %s", err)
+		return nil, fmt.Errorf("opening datastore on repo: %s", err)
 	}
 	ip2l := ip2location.New([]string{"./ip2location-ip4.bin"})
 
 	ai, err := ask.New(txndstr.Wrap(ds, "index/ask"), c)
 	if err != nil {
-		return nil, fmt.Errorf("error when creating ask index: %s", err)
+		return nil, fmt.Errorf("creating ask index: %s", err)
 	}
 	mi, err := miner.New(txndstr.Wrap(ds, "index/miner"), c, fchost, ip2l)
 	if err != nil {
-		return nil, fmt.Errorf("error when creating miner index: %s", err)
+		return nil, fmt.Errorf("creating miner index: %s", err)
 	}
 	si, err := slashing.New(txndstr.Wrap(ds, "index/slashing"), c)
 	if err != nil {
-		return nil, fmt.Errorf("error when creating slashing index: %s", err)
+		return nil, fmt.Errorf("creating slashing index: %s", err)
 	}
 	dm, err := deals.New(c, deals.WithImportPath(filepath.Join(conf.RepoPath, "imports")))
 	if err != nil {
-		return nil, fmt.Errorf("error when creating deal module: %s", err)
+		return nil, fmt.Errorf("creating deal module: %s", err)
+	}
+	wm, err := wallet.New(c, &masterAddr, conf.WalletInitialFunds)
```

Comment:
The Wallet Module is configured now with a _masterAddr_ and a _WalletInitialFunds_.
Every new wallet created with be sent that amount of _attoFIL_ amount from _masterAddr_.
If _masterAddr_ is nil, then no funds will be sent.

Most probably this will change a bit more in the future since it may be better to configure this action _per wallet creation_ and not globally.